ttt.c

```diff
@@ -10,16 +10,26 @@
 
 #define MAX_SEARCH_DEPTH 6
 #define BOARD_SIZE 4
+#define GOAL_NUMBER 3
 #define N_GRIDS (BOARD_SIZE) * (BOARD_SIZE)
 #define GET_INDEX(i, j) (i) * (BOARD_SIZE) + (j)
 #define GET_COL(x) ((x) % (BOARD_SIZE))
 #define GET_ROW(x) ((x) / (BOARD_SIZE))
 
 _Static_assert(BOARD_SIZE < 26, "Board size must be less than 26");
 
-enum { ROW, COL, PRIMARY, SECONDARY };
-static int no_step[2] = {0, 0};
-static const int step_forward[2] = {1, 2}, step_backward[2] = {-1, -2};
+typedef struct {
+    int i_shift, j_shift, i_lower_bound, j_lower_bound, i_upper_bound,
+        j_upper_bound;
+} line_t;
+static line_t lines[4] = {
+    {1, 0, 0, 0, BOARD_SIZE - GOAL_NUMBER + 1, BOARD_SIZE},  // ROW
+    {0, 1, 0, 0, BOARD_SIZE, BOARD_SIZE - GOAL_NUMBER + 1},  // COL
+    {1, 1, 0, 0, BOARD_SIZE - GOAL_NUMBER + 1,
+     BOARD_SIZE - GOAL_NUMBER + 1},  // PRIMARY
+    {1, -1, 0, GOAL_NUMBER - 1, BOARD_SIZE - GOAL_NUMBER + 1,
+     BOARD_SIZE}  // SECONDARY
```

Comment:
Place a comma `,` right after `BOARD_SIZE}`, so that clang-format can indent better.